tests/test_xopen.py

```diff
@@ -118,6 +118,16 @@ def test_pipedgzipreader_textiowrapper():
             assert wrapped.read() == CONTENT
 
 
+def test_detect_gzip_file_format_from_content():
+    with xopen("tests/file.txt.gz.test", "rb") as fh:
```

Comment:
Why are the tests opening plaintext files? That will always succeed, but not touch the code that you wrote.